src/agent/storage-queue/src/lib.rs

```diff
@@ -145,11 +158,17 @@ impl Message {
     }
 
     pub fn get<'a, T: serde::de::Deserialize<'a>>(&'a self) -> Result<T> {
-        let data = serde_json::from_slice(&self.data)?;
+        let data =
+            serde_json::from_slice(&self.data).context("get storage queue message failed")?;
         Ok(data)
     }
 }
 
+fn is_empty_message(text: &str) -> bool {
+    text.contains(r"<QueueMessagesList></QueueMessagesList>")
```

Comment:
Would be good to add a comment with this link: `https://docs.microsoft.com/en-us/rest/api/storageservices/get-messages#response-body`

This check is a bit brittle, it'd be nice if we could handle these cases:
```
<QueueMessagesList>  </QueueMessagesList>
```
```
<QueueMessagesList>
</QueueMessagesList>
```